### ISSUE TITLE
Support Amazon's new Nova models on bedrock

### ISSUE THREAD

**Issue by cmaycumber** [vercel/ai#4087]
### Feature Description

Add support (or documentation for the new Nova line of foundational models)

**Comment by lgrammel** [comment]
Can you just enter them as model ids? (they should work out of the box)

**Comment by pr1ntr** [comment]
that yeilds this error: 
```
ValidationException: This model doesn't support the toolConfig.toolChoice.tool field.
```

with this code

```
const { object } = await generateObject({
    model,
    temperature: 0.8,
    maxTokens: 5120,
    output: 'object',
    schema,
    ...(system ? { system } : {}),
    messages: [
      {
        role: 'user',
        content: [
          {
            type: 'text',
            text: prompt,
          },
          ...fileParts,
        ],
      },
    ],
  })
```

**Comment by pr1ntr** [comment]
Is there any update on this? I think the issue is particular to `generateObject`

**Comment by samatcd** [comment]
Yes, this appears to only be affecting `generateObject`, and looks to be due to this method using tool calling under-the-hood. The tool call by the provider automatically adds `"toolChoice": { "tool": { "name": "json" } }` to the `toolConfig` which causes a `ModelErrorException` to be returned by the bedrock service.

Adding `toolChoice: 'required'` to `generateText` calls also causes this same error.

`toolChoice` for `generateObject` in this provider is added here. If using Nova model, this should be omitted:
https://github.com/vercel/ai/blob/f53c8cda8de6083869d6f06cd349fa51d9da92b8/packages/amazon-bedrock/src/bedrock-chat-language-model.ts#L167

**Comment by samatcd** [comment]
Further to this, per Nova's tool-use documentation:

> Amazon Nova only supports the auto option for toolChoice, which is the default. Trying to set any or specific tool names will cause a validation error.

https://docs.aws.amazon.com/nova/latest/userguide/tool-use-definition.html#:~:text=Amazon%20Nova%20only%20supports%20the,will%20cause%20a%20validation%20error.&text=The%20name%2C%20description%2C%20and%20the,exact%20functionality%20of%20the%20tool.

**Comment by dreamorosi** [comment]
I can confirm that manually patching out the line mentioned above makes the Nova models work with `generateObject`. While working, the results are mixed though - but this is a model-related thing.

**Comment by dreamorosi** [comment]
AWS added support for tool choice as announced [here](https://aws.amazon.com/about-aws/whats-new/2025/03/amazon-nova-expands-tool-converse-api/):
> With this update, Nova adds support for 'Any' and 'Tool' modes in addition to the existing 'Auto' mode support

I have tested this code with these versions of `ai@4.2.2` and `@ai-sdk/amazon-bedrock@2.2.1` and was able to generate an object successfully:

```ts
import { bedrock } from '@ai-sdk/amazon-bedrock';
import { generateObject } from 'ai';
import { z } from 'zod';

const res = await generateObject({
  model: bedrock('us.amazon.nova-lite-v1:0'),
  schema: z.object({
    recipe: z.object({
      name: z.string(),
      ingredients: z.array(z.object({ name: z.string(), amount: z.string() })),
      steps: z.array(z.string()),
    }),
  }),
  prompt: 'Generate a lasagna recipe.',
});
console.log(JSON.stringify(res.object, null, 2));
```

I think this issue can be closed @lgrammel.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The discussion identified a concrete compatibility failure in existing `generateObject` behavior, despite the original support request and the later upstream AWS change resolving it.

### Evidence
- Nova model IDs were expected to work out of the box, but `generateObject` returned a Bedrock `ValidationException` because the SDK supplied a named `toolChoice`.
- AWS documentation initially stated Nova supported only automatic tool choice; manually removing the SDK-added tool choice made `generateObject` work.
- A later AWS update added the required tool-choice modes, and a commenter confirmed successful object generation with current package versions.
